### ISSUE TITLE
Can not compile in Linux

### ISSUE THREAD

**Issue by msmalik681** [rofl0r/obeditor#2]
I am having trouble compiling this in Linux. Tested in Lubuntu 16.04 LTS installed libgif 4 and wxWidgets-WX_2_8_12 the compiler has warnings but the error is reguading libpng12.so.0 any idea how to fix this ?

**Comment by rofl0r** [comment]
> the error

what is the error you're getting?
according to scons file libpng shouldnt be required at all, but its possible that wxwidgets depends on it, however it should use libpng 1.5+, not 1.2.

**Comment by msmalik681** [comment]
This must have been the issue I presumed a older version of libpng was required.  My virtual machine broke trying to move from libpng12 to libpng16 maybe the system was  too dependant on 12 I will test again tomorrow.

**Comment by msmalik681** [comment]
Just setup ubuntu 18.04 it comes with libpng 1.6. Setup with wxwidgets 2_9_4 and libgif 4.2.3 but still getting png related errors.

```
. . .

```

**Comment by rofl0r** [comment]
should work now (pull changes), i got it to compile with wxGTK-2.8.12

**Comment by msmalik681** [comment]
Tried the updated version now I am getting errors about gif lib but I installed giflib-4.2.3 and I am using gcc/g++ 4.8.5 with wxGTK-2.8.12. Can I ask what OS you are compiling on and what exact versions of libgif and gcc/g++ are you using.  I have been trying for days with no luck.

**Comment by rofl0r** [comment]
giflib-5.2.1.tar.gz
libpng-1.5.30.tar.xz
gcc 6.5.0

i compiled it on my own OS called sabotage linux, which is built from source

**Comment by msmalik681** [comment]
@rofl0r Ok so you have proven this is a issue from my end I can not satisfy the required dependencies so I guess this issue can be closed. This application would be easier to build if it was using wxgtk3.0 it would not be a huge change to update the code I tried myself and failed I did manage to compile (with errors) but the executable would not start and crashed with errors.

I wanted to make a branch with my effort for you to review but I can not make a branch I think you need to grant me access. For now here is the attempt I made to update to wxgtk3.0: https://www.mediafire.com/file/2h6c2zz5y1wxvsn/obeditor-master-wxgtk3.0.tar.bz2/file

There was one function I think to confirm a valid character was typed but it was causing errors and I did not know how to fix it so I commented it out and left a todo note.

**Comment by rofl0r** [comment]
> I can not satisfy the required dependencies

it's supposed to work with latest libpng and giflib.
if you share the error message you get i could help.

> easier to build if it was using wxgtk3.0

i'm not interested in GTK+3, all my apps run on GTK+2. see [my article ](https://sabotage-linux.neocities.org/blog/8/) about the topic.

**Comment by rofl0r** [comment]
i added 2 commits, one fixing a scons py3 compatibility issue, one adding build instructions for ubuntu 20.04.
using these instructions and ubuntu vanilla dependencies of giflib and png, my build succeeded.

**Comment by rofl0r** [comment]
> I wanted to make a branch with my effort for you to review but I can not make a branch I think you need to grant me access

you need to click the "fork" button on this repo then push your changes to your own fork.

**Comment by msmalik681** [comment]
Oh my god you Legend! Thank you for your patience with me. I followed your instructions and it worked first time. I was compiling the source with simple "./configure make install" and I was getting some narrowing error but you fixed it.

I have another question how can I compile the windows version on Linux.

Also if I make my own fork and make changes do I then make a pull request so you can update the the code here? I am not a very experienced coder and I do not know anything about wxwidgets yet but I would like to help update this project in the future. I will be converting this to appimage format can I add it to the releases here and would this still be version 1.5?

**Comment by rofl0r** [comment]
> I was compiling the source with simple "./configure make install" and I was getting some narrowing error but you fixed it.

yeah, that's the special appeal of C++: they release a new standard every 2 years and all developers are eager to use the new features, so compilers make the new standard the default quite quickly, thereby breaking older C++ code when compiled with default setting.

> I have another question how can I compile the windows version on Linux.

that's gonna be real hard. i think the easiest method is to install cygwin on windows and compile the things inside cygwin - you need to compile all dependencies that cygwin doesnt provide from source too.

if you ultimate goal is to have an openbor editor for windows, i'd suggest to use https://github.com/OpenBOR/OpenBORStats instead - it's a lot better, much more functionality, much cleaner code and is actually hackable. however it requires borland delphi. in fact this editor here heavily uses OOP inheritance which makes the codeflow almost impossible to understand. at least i myself struggled when i tried to add some features once.

> Also if I make my own fork and make changes do I then make a pull request so you can update the the code here?

that's how it's usually done, indeed.

> I will be converting this to appimage format can I add it to the releases here and would this still be version 1.5?

we can talk about the intrinsics of this once you get there, but yeah something like that could be done.

**Comment by msmalik681** [comment]
Thank you for your input I appreciate all the support and I really just want a OpenBOR editor that is cross platform (Windows and Linux not really concerned with MAC) and that can be compiled from source so updating is not out of the question. I think matmans editor was windows only. The other candidate uses QT5 and Python here is the source code: https://www.mediafire.com/file/5ivwj9djhwv6n2n/chronocrash-modders-tools-src2.zip/file I have no idea how to compile it or if it works on Linux I am trying to contact the developer but he is hard to get hold of. No Linux support is a deal breaker for me I know about WINE but still need native support.

Made the Appimage for Linux that should run on almost all distros. If you want to add it to releases it is here: https://www.mediafire.com/file/d38dh6jh8zk36f5/OpenBOR_Editor_1.5_Linux.zip/file
